website/src/content/migrating-from-project-level-custom-applications.mdx

```diff
@@ -0,0 +1,271 @@
+
+Upgrade your Application Kit packages (`@commercetools-frontend/*`) to version `>= 21`. Follow the instructions in the [Release notes](/releases/2022-01-25-custom-applications-v21).
+
+<Info>
+
+If you need to upgrade from several major versions behind, we recommend to go through the [migration guides](https://docs.commercetools.com/custom-applications/releases?filterTopics=Dependencies) of each major version first.
+
+</Info>
+
+## Update the Custom Application config
+
+The Custom Application config has been enhanced with more required fields:
+
+```json title="custom-application-config.json" highlightLines="6-8,10,14-31"
+{
+  "name": "Avengers app",
+  "entryPointUriPath": "avengers",
+  "cloudIdentifier": "gcp-eu",
+  "env": {
+    "development": {
+      "initialProjectKey": "<project_key>"
+    },
+    "production": {
+      "applicationId": "<application_id>",
+      "url": "https://avengers.app"
+    }
+  },
+  "oAuthScopes": {
+    "view": ["view_states", "view_products"],
+    "manage": ["manage_products"]
+  },
+  "icon": "${path:@commercetools-frontend/assets/application-icons/rocket.svg}",
+  "mainMenuLink": {
+    "defaultLabel": "Avengers",
+    "labelAllLocales": [],
+    "permissions": [],
+  },
+  "submenuLinks": [
+    {
+      "uriPath": "new",
+      "defaultLabel": "Add Avenger",
+      "labelAllLocales": [],
+      "permissions": [],
+    }
+  ]
+}
+```
+
+See [Custom Application config](/api-reference/application-config) for more information.
+
+
+## Choose the `entryPointUriPath`
+
+The `entryPointUriPath` plays a crucial role in a Custom Application. Effectively, it's the **identifier** of the Custom Application similarly to the `projectKey` of the commercetools Platform.
+
+<Info>
+
+From now on, with the introduction of Org-level Custom Applications, the `entryPointUriPath` is **considered unique** within each [cloud region](https://docs.commercetools.com/api/general-concepts#regions) environment, meaning that there can't be any duplicates.
+
+</Info>
+
+Choosing the `entryPointUriPath` becomes then more important and should be done carefully as it affects routing and permissions.
+
+The `entryPointUriPath` value must adhere to the following restrictions:
+
+* Only lowercase alphanumeric characters are allowed.
+* Must be between 2 and 64 characters.
+* Underscores and hyphens are allowed except as leading, trailing, and adjacent characters.
+
+Furthermore, the `entryPointUriPath` value is bound to User Permissions of the Custom Application and is used to derive the unique name of the User Permissions.<br />
+The name consists of the following format: `{View,Manage}<EntryPointUriPath>`.
+
+Here are some examples:
+
+| `entryPointUriPath` | User Permission |
+| --- | --- |
+| `avengers` | `{View,Manage}Avengers` |
+| `the-avengers` | `{View,Manage}TheAvengers` |
+| `the_avengers` | `{View,Manage}The_Avengers` |
+| `avengers-01` | `{View,Manage}Avengers/01` |
+| `avengers_01` | `{View,Manage}Avengers_01` |
+
```

Comment:
I am wondering if we could use a brand free example. I am worried that we could get in trouble using "Avengers".